packages/datadog-plugin-cucumber/src/index.js

```diff
@@ -0,0 +1,95 @@
+const { relative } = require('path')
+
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+
+const {
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS,
+  getTestEnvironmentMetadata
+} = require('../../dd-trace/src/plugins/util/test')
+
+function setStatusFromResult (span, result) {
+  if (result.status === 1) {
+    span.setTag(TEST_STATUS, 'pass')
+  } else if (result.status === 2) {
+    span.setTag(TEST_STATUS, 'skip')
+    span.setTag('error.msg', 'skipped')
+  } else if (result.status === 4) {
+    span.setTag(TEST_STATUS, 'skip')
+    span.setTag('error.msg', 'not implemented')
+  } else {
+    span.setTag(TEST_STATUS, 'fail')
+    span.setTag('error.msg', result.message)
+  }
+}
+
+function createWrapRun (tracer, testEnvironmentMetadata, sourceRoot) {
+  return function wrapRun (run) {
+    return async function handleRun (...args) {
+      const testName = this.pickle.name
+      const testSuite = relative(sourceRoot, this.pickle.uri)
+
+      const commonSpanTags = {
+        [TEST_TYPE]: 'test',
+        [TEST_NAME]: testName,
+        [TEST_SUITE]: testSuite,
+        [SAMPLING_RULE_DECISION]: 1,
+        ...testEnvironmentMetadata
+      }
+
+      const result = await tracer.trace(
+        'cucumber.test',
+        {
+          type: 'test',
+          resource: testName,
+          tags: commonSpanTags
+        },
+        async (span) => {
+          const runResult = await run.apply(this, args)
+          setStatusFromResult(span, this.getWorstStepResult())
+          return runResult
```

Comment:
in the case of `jest` I need to do the `tracer.wrap` dance because I need to instrument a parameter of the function (`event.test.fn`), but in this case since you're instrumenting the function itself, maybe it can be simplified? 

```javascript
return async function handleRun (...args) {
      const testName = this.pickle.name
      const testSuite = relative(sourceRoot, this.pickle.uri)

      const commonSpanTags = {
        [TEST_TYPE]: 'test',
        [TEST_NAME]: testName,
        [TEST_SUITE]: testSuite,
        [SAMPLING_RULE_DECISION]: 1,
        ...testEnvironmentMetadata
      }
      const span = tracer.scope().active()
      span.addTags(commonSpanTags)
      const result = await run.apply(this, arguments)
      setStatusFromResult(span, this.getWorstStepResult())
      return result
```
I'm not sure but in that case you might have to bind the scope
```javascript
tracer.scope().bind(run, span).apply(this, arguments)
```

Reply:
I want to create a new span for step. I am not sure if `const span = tracer.scope().active()` will do that.